packages/http-server/test/integration/http-server.integration.ts

```diff
@@ -0,0 +1,112 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/http-server
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {HttpServer} from '../../';
+import {supertest, expect} from '@loopback/testlab';
+import * as makeRequest from 'request-promise-native';
+
+describe('HttpServer (integration)', () => {
+  it('starts server', async () => {
+    const server = new HttpServer(
+      (req, res) => {
+        res.end();
+      },
+      {port: 0},
```

Comment:
Would it make sense to make the `port` setting optional and defaulting to `0`?

Intended usage:

```ts
const server = new  HttpServer(requestHandler);
```

It would also allow us to simplify `RestServerConfig` - instead of copying all interface members, we could leverage inheritance.

```ts
interface RestServerConfig extends HttpServerConfig {
  // ...
}
```

Thoughts?